presto-spark-base/src/main/java/com/facebook/presto/spark/planner/PrestoSparkRddFactory.java

```diff
@@ -220,7 +222,8 @@ private <T extends PrestoSparkTaskOutput> JavaPairRDD<MutablePartitionId, T> cre
                 outputType);
 
         Optional<PrestoSparkTaskSourceRdd> taskSourceRdd;
-        List<TableScanNode> tableScans = findTableScanNodes(fragment.getRoot());
+        Optional<NativeExecutionNode> nativeExecutionNode = findNativeExecutionNodes(fragment.getRoot());
```

Comment:
We probably will need some Spark experts to review the Spark related code to guard the quality. I'm not sure if James is familiar with this piece of code though.

Reply:
The changes in the Rdd factory looks a bit hacky and dirty to me. I don't know too much about Spark rdd but maybe we can have an offline discussion on this to understand better.

Reply:
Refactored this part based on @rschlussel and your feedbacks, could you take another look? Thanks

Reply:
Thanks that looks a lot cleaner.